pkg/storage/ingest/writer.go

```diff
@@ -195,7 +202,7 @@ func (w *Writer) newKafkaWriter(partitionID int32) (*kgo.Client, error) {
 		kgo.DefaultProduceTopic(w.kafkaCfg.Topic),
 
 		// Use a static partitioner because we want to be in control of the partition.
-		kgo.RecordPartitioner(newKafkaStaticPartitioner(int(partitionID))),
+		kgo.RecordPartitioner(newKafkaStaticPartitioner()),
```

Comment:
kgo has `ManualPartitioner()` which does the same as the `newKafkaStaticPartitioner()` now

Reply:
You're absolute right. Done in https://github.com/grafana/mimir/pull/8088/commits/32c5ba5ab12f931e2d171d9afee4a070398739e6.